src/main/scala/org/constellation/infrastructure/endpoints/SnapshotEndpoints.scala

```diff
@@ -192,12 +192,17 @@ class SnapshotEndpoints[F[_]](implicit F: Concurrent[F]) extends Http4sDsl[F] {
             case Invalid(IncorrectSenderId(_))                                                  => Response[F](status = Unauthorized).pure[F]
             case Invalid(e)                                                                     => logger.error(e)(e.getMessage) >> InternalServerError()
             case Valid(_) =>
-              for {
-                _ <- redownloadService.persistPeerProposal(message.origin, payload.proposal)
-                _ <- redownloadService.updatePeerMajorityInfo(message.origin, payload.majorityInfo)
-                _ <- snapshotProposalGossipService.spread(message)
-                res <- Ok()
-              } yield res
+              payload.proposal.validSignature
+                .pure[F]
+                .ifM(
+                  for {
+                    _ <- redownloadService.persistPeerProposal(message.origin, payload.proposal)
+                    _ <- redownloadService.updatePeerMajorityInfo(message.origin, payload.majorityInfo)
+                    _ <- snapshotProposalGossipService.spread(message)
```

Comment:
BTW are there context shifts to other thread pools inside of these functions? I wonder where should these computations run...

Reply:
Actually, that's a good question. I just assumed that there are context shifts, but as I look into the code I can't find any. That would mean, that the consecutive calls on the gossip path are nested (so doing bisect doesn't really make much sense). @kpudlik can you confirm my findings?

Reply:
@TheMMaciek I will test it on cluster to check if it is the case that there are no context shifts on `spread`. If it will be the case I will raise another PR. Would that be fine with you?

Reply:
Yeah, sure. I guess it's good to be aware where stuff is running and adjusting if needed.

Reply:
Great, so I'm merging this one.